sdk/pinocchio/src/pubkey.rs

```diff
@@ -232,3 +234,49 @@ pub fn checked_create_program_address(
 
     create_program_address(seeds, program_id)
 }
+
+/// Derive a Pubkey from another Pubkey, seed, and a program id.
+#[inline]
+pub fn create_with_seed(
+    base: &Pubkey,
+    seed: &[u8],
+    owner: &Pubkey,
+) -> Result<Pubkey, ProgramError> {
+    if seed.len() > MAX_SEED_LEN {
+        return Err(ProgramError::MaxSeedLengthExceeded);
+    }
+
+    if owner.len() >= PDA_MARKER.len() {
+        let slice = &owner[owner.len() - PDA_MARKER.len()..];
+        if slice == PDA_MARKER {
+            return Err(ProgramError::IllegalOwner);
+        }
+    }
```

Comment:
Is this the same as `ends_with()`?

```suggestion
    if owner.ends_with(PDA_MARKER) {
        return Err(ProgramError::IllegalOwner);
    }
```